src/doc/trpl/error-handling.md

```diff
@@ -1,308 +1,2147 @@
+* [The short story](#the-short-story)
+
+## The Basics
+
+You can think of error handling as using *case analysis* to determine whether
+a computation was successful or not. As you will see, the key to ergonomic error
+handling is reducing the amount of explicit case analysis the programmer has to
+do while keeping code composable.
+
+Keeping code composable is important, because without that requirement, we
+could [`panic`](http://doc.rust-lang.org/std/macro.panic!.html) whenever we
+come across something unexpected. (`panic` causes the current task to unwind,
+and in most cases, the entire program aborts.) Here's an example:
 
-> The best-laid plans of mice and men  
-> Often go awry
->
-> "Tae a Moose", Robert Burns
+```rust
+// Guess a number between 1 and 10.
+// If it matches the number we had in mind, return true. Else, return false.
+fn guess(n: i32) -> bool {
+    if n < 1 || n > 10 {
+        panic!("Invalid number: {}", n);
+    }
+    n == 5
+}
+
+fn main() {
+    guess(11);
+}
+```
+
+If you try running this code, the program will crash with a message like this:
+
+```
+thread '<main>' panicked at 'Invalid number: 11', src/bin/panic-simple.rs:5
+```
+
+Here's another example that is slightly less contrived. A program that accepts
+an integer as an argument, doubles it and prints it.
+
+```rust
+use std::env;
+
+fn main() {
+    let mut argv = env::args();
+    let arg: String = argv.nth(1).unwrap(); // error 1
+    let n: i32 = arg.parse().unwrap(); // error 2
+    println!("{}", 2 * n);
+}
+
+// $ cargo run --bin unwrap-double 5
+// 10
+```
+
+If you give this program zero arguments (error 1) or if the first argument
+isn't an integer (error 2), the program will panic just like in the first
+example.
 
-Sometimes, things just go wrong. It's important to have a plan for when the
-inevitable happens. Rust has rich support for handling errors that may (let's
-be honest: will) occur in your programs.
+You can think of this style of error handling as similar to a bull running
+through a china shop. The bull will get to where it wants to go, but it will
+trample everything in the process.
 
-There are two main kinds of errors that can occur in your programs: failures,
-and panics. Let's talk about the difference between the two, and then discuss
-how to handle each. Then, we'll discuss upgrading failures to panics.
+### Unwrapping explained
 
-# Failure vs. Panic
+In the previous example, we claimed
+that the program would simply panic if it reached one of the two error
+conditions, yet, the program does not include an explicit call to `panic` like
+the first example. This is because the
+panic is embedded in the calls to `unwrap`.
 
-Rust uses two terms to differentiate between two forms of error: failure, and
-panic. A *failure* is an error that can be recovered from in some way. A
-*panic* is an error that cannot be recovered from.
+To "unwrap" something in Rust is to say, "Give me the result of the
```

Comment:
We currently use curly quotes in the book. If you don't want to convert these, I can send a PR to your PR :)